driver/src/orderbook/batched_auction_data_reader.rs

```diff
@@ -0,0 +1,288 @@
+use crate::contracts::stablex_auction_element::{StableXAuctionElement, AUCTION_ELEMENT_WIDTH};
+use dfusion_core::models::{AccountState, BatchInformation, Order};
+use std::collections::HashMap;
+use web3::types::{H160, U256};
+
+/// Handles reading of auction data that has been encoded with the smart
+/// contract's `encodeAuctionElement` function.
+pub struct BatchedAuctionDataReader {
+    /// The account state resulting from unfiltered handled orders.
+    pub account_state: AccountState,
```

Comment:
Shouldn't all of these fields be read only? We don't want anyone to mutate the reader's state.